test/e2e/framework/kcp.go

```diff
@@ -337,7 +335,7 @@ func (c *kcpServer) monitorEndpoint(client *rest.RESTClient, endpoint string) {
 	}
 	wait.UntilWithContext(ctx, func(ctx context.Context) {
 		_, err := rest.NewRequest(client).RequestURI(endpoint).Do(ctx).Raw()
-		if c.ctx.Err() != nil {
+		if errors.Is(err, context.Canceled) {
```

Comment:
I'm not sure this is right - if the context controlling the monitor goroutine is cancelled, we should abort doing _anything_. The `errors.Is()` check makes sense on the consumer end, not the producer?

Reply:
Oh thanks, yeah I missed they were 2 different contexts

Reply:
How about `if errors.Is(c.ctx.Err(), context.Canceled) {`

Reply:
I don't think we need to care if the context was cancelled or timed out?

Reply:
From the doc:

```
	// If Done is not yet closed, Err returns nil.
	// If Done is closed, Err returns a non-nil error explaining why:
	// Canceled if the context was canceled
	// or DeadlineExceeded if the context's deadline passed.
	// After Err returns a non-nil error, successive calls to Err return the same error.
	Err() error
```

Here we just care to see if it's done, for any reason

Reply:
Ok I changed it back but it’s flaking again. I have a theory:
1. ctx is first set to c.ctx
1. It’s then changed to a context w/a deadline
1. c.ctx is NOT updated w/a deadline (appropriate)
1. When the test completes, its cleanup function cancels the deadline context
1. The ctx passed into wait.UntilWithContext is the deadline context
1. Test cleanup happens -> deadline context is canceled
1. rest request returns an error, which is context.Canceled, but c.ctx hasn’t been canceled (yet)